zookeeper-server/src/main/java/org/apache/zookeeper/server/DataTree.java

```diff
@@ -602,11 +584,11 @@ public void deleteNode(String path, long zxid)
         String lastPrefix = getMaxPrefixWithQuota(path);
```

Comment:
(GitHub won't let me add a note on line 577—about something I noticed but which was not introduced by this patch.)

Isn't this code slightly wrong?

```java
        if (parentName.startsWith(procZookeeper) && Quotas.limitNode.equals(childName)) {
            // delete the node in the trie.
            // we need to update the trie as well
            pTrie.deletePath(parentName.substring(quotaZookeeper.length()));
        }
```

Shouldn't it test for `parentName.startsWith(quotaZookeeper)`, instead of `procZookeeper`?

Oh, and `DataTree` is full of `.substring(quotaZookeeper.length())` which could be replaced by your `trimQuotaPath` helper :)

Reply:
- Yes, ` parentName.startsWith(quotaZookeeper)` is closer to the truth. we can create another ticket to fix it
- substring(quotaZookeeper.length())  are all replaced with trimQuotaPath